hyperhtml.js

```diff
@@ -193,321 +156,692 @@ var hyperHTML = (function () {'use strict';
+  var templateObjects, unique;
+  if (FF) {
+    templateObjects = Object.create(null);
+    unique = function (template) {
+      var key = template.join(UID);
+      return templateObjects[key] ||
+            (templateObjects[key] = template);
+    };
+  }
+
+  // use native .append(...childNodes) where available
+  var appendNodes = 'append' in document ?
+      function (node, childNodes) {
+        node.append.apply(node, childNodes);
+      } :
+      function appendNodes(node, childNodes) {
+        for (var
+          i = 0,
+          length = childNodes.length;
+          i < length; i++
+        ) {
+          node.appendChild(childNodes[i]);
+        }
+      };
+
+  // returns children or retrieve them in IE/Edge
+  var getChildren = 'children' in document ?
+      function (node) { return node.children; } :
+      function (node) {
+        for (var
+          child,
+          children = [],
+          childNodes = node.childNodes,
+          j = 0, i = 0, length = childNodes.length;
+          i < length; i++
+        ) {
+          child = childNodes[i];
+          if (child.nodeType === ELEMENT_NODE)
+            children[j++] = child;
+        }
+        return children;
+      };
+
+  // return the correct node walking through a path
+  // fixes IE/Edge issues with attributes and children
+  var getNode = IE ?
+      function (parentNode, path) {
+        for (var name, i = 0, length = path.length; i < length; i++) {
+          name = path[i++];
+          switch (name) {
+            case 'children':
+              parentNode = getChildren(parentNode)[path[i]];
+              break;
+            case GET_ATTRIBUTE_NODE:
+              parentNode = parentNode[name](path[i]);
+              break;
+            default:
+              parentNode = parentNode[name][path[i]];
+              break;
+          }
+        }
+        return parentNode;
+      } :
+      function (parentNode, path) {
+        for (var i = 0, length = path.length; i < length; i++) {
+          parentNode = parentNode[path[i++]][path[i]];
+        }
+        return parentNode;
+      };
+
+  // fixes IE problems with comments
+  if (IE) {
+    var IEAttributes;
+    var no = new RegExp('([^\\S][a-z]+[a-z0-9_-]*=)([\'"])' + UIDC + '\\2', 'g');
+    var comments = function ($0, $1, $2) {
+          IEAttributes.push($1.slice(1, -1));
+          return $1 + $2 + UID + $2;
+        };
+  }
+
+  // [element] = {template, updates};
+  var hypers = new $WeakMap;
+
+  // [element] = {template, updates};
+  var wires = new $WeakMap;
+
+  // [template] = {fragment, paths};
+  var templates = new $Map;
+
+  // internal signal to switch adoption
+  var notAdopting = true;
+
+  // ---------------------------------------------
+  // Template related utilities
+  // ---------------------------------------------
+
+  // given a unique template object
+  // create, parse, and store retrieved info
+  function createTemplate(template) {
```

Comment:
Hey @marcoscaceres I've actually rush-replied 'cause I was in the middle of other changes so please don't get my comments wrong, I'm just trying to explain.

Meanwhile I've tried to follow all your hints and put comments everywhere to avoid confusion or be scared so thanks a lot for the effort!